lib/vtls/mbedtls.c

```diff
@@ -46,12 +46,12 @@
 #include <mbedtls/x509.h>
 #include <mbedtls/psa_util.h>
 
-#if MBEDTLS_VERSION_NUMBER < 0x04000000 && defined(MBEDTLS_CTR_DRBG_C)
-#define CURL_MBEDTLS_DRBG
+#if MBEDTLS_VERSION_NUMBER < 0x04000000 && !defined(MBEDTLS_CTR_DRBG_C)
+#error "MBEDTLS_CTR_DRBG_C is required for mbedTLS 3.x and older."
```

Comment:
curl only supports 3.x+, so this in effect means dropping support for
mbedTLS 3.x without the `MBEDTLS_CTR_DRBG_C` build option.

If so, It's preferred to do delete features separate from other changes.
Also explaining this in the PR message would be important.

Mixed with 8 other changes, it's difficult to say which other changes
are related to this.